dotnet/src/SemanticKernel.Abstractions/Contents/FunctionCallContentBuilder.cs

```diff
@@ -181,13 +181,13 @@ private static void TrackStreamingFunctionCallUpdate(StreamingFunctionCallUpdate
 
         // If we have an call id, ensure the index is being tracked. Even if it's not a function update,
         // we want to keep track of it so we can send back an error.
-        if (update.CallId is string id)
+        if (update.CallId is string id && !string.IsNullOrEmpty(id))
```

Comment:
I wonder if we should append instead of setting the name.

For arguments we append, if we append `empty` or `null` we won't lose information and will be compatible with multi chunk namings.

Reply:
Yeah, I had the same thought. I did some investigation and the function name is limit to 64 chars and doesn't seem to be streamed. This is the sample on the OpenAI forum https://community.openai.com/t/functions-calling-with-streaming/305742. note that it doesn't append the function name or if but it does the function arguments. We haven't had any feedback that developers are running into issues with streamed function names so I'm going to leave as is until we get information that this behaviour is needed.